### PR TITLE
Terrain details pipeline resources 19.1 backport

### PR DESCRIPTION
**Pull request by russoh**
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?

Backport for https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/2550
It will require engine C++ code to be backported before this can be merged. Please do no merge until go ahead is given.

---
### Release Notes
Please add any useful notes about the feature/fix that might be helpful for others.

---
### Testing status
**Katana Tests**: https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=terrain-details-pipeline-resources-19.1&automation-tools_branch=master&unity_branch=2019.1%2Fstaging#

**Manual Tests**: What did you do?

**Automated Tests**: What did you setup?

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?

**Halo Effect**: None, Low, Medium, High?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### Added
 - Added software dynamic resolution support
+- Added support for overriding terrain detail rendering shaders, via the render pipeline editor resources asset
```
Comment (Katstodian) [inline com.unity.render-pipelines.high-definition/CHANGELOG.md]: Please change to:
Added support for overriding Shaders that render terrain details. You can override them via the render pipeline editor resources asset.

**Comment by russoh** [comment]
Green Katana tests using this branch with engine backport branch:
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=terrain-details-pipeline-resources-19.1&automation-tools_branch=add-platform-filter&unity_branch=2019.1%2Fgraphics%2Fterrain-details-asset-hooks

**Comment by Verasl** [comment]
This is now a high priority as the changes landed in 19.1 Unity side and has broken tests

**Comment by sebastienlagarde** [comment]
Merged master and solve the conflict

**Comment by sebastienlagarde** [comment]
Katana is green. Merging

**Pull request merged**